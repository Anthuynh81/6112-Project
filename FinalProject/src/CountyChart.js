import React, {memo, useEffect, useState} from "react";
import {
    ZoomableGroup,
    ComposableMap,
    Geographies,
    Geography
} from "react-simple-maps";
import Axios from 'axios';
import ReactTooltip from "react-tooltip";

const geoUrl = "https://cdn.jsdelivr.net/npm/us-atlas@3/counties-10m.json";
const startDate = new Date(2020,0,23);

const CountyChart = (props) => {
    const [data, setData] = useState([]);
    const [loading, setLoading] = useState(true);
    const [date, setDate] = useState(startDate.toLocaleDateString("en-US", {
        year: "numeric",
        month: "2-digit",
        day: "2-digit",
    }));

    if (props.date !== date){
        setLoading(true);
        setDate(props.date);
    }

    useEffect(() => {
        Axios.get("http://localhost:5000/County", { params: {date: date}}).then((response) =>{
            setData(response.data)
            setLoading(false);
            });
    }, [date]);

    if(props.map === "County"){
        return (
            <>
                <ComposableMap data-tip="" projection="geoAlbersUsa" width={800} height={600}>
                    <ZoomableGroup zoom={1}>>
                        <Geographies geography={geoUrl}>
                            {({ geographies }) =>
                                geographies.map(geo => (
                                    <Geography
                                        key={geo.rsmKey}
                                        geography={geo}
                                        onMouseEnter={() => {
                                            const NAME = geo.properties.name;
                                            const state = data.find(o => o.AdminRegion2 === NAME);
                                            if(loading){
                                                props.setTooltipContent('Data is loading')
                                            }else{
                                                if (state){
                                                    props.setTooltipContent(`${geo.properties.name}</br>
                                                        Confirmed - ${state.Confirmed}</br>`);
                                                } else {
                                                    props.setTooltipContent(`${geo.properties.name}</br>
                                                        Confirmed - 0</br>`);
                                                }

                                            }
                                        }}
                                        onMouseLeave={() => {
                                            props.setTooltipContent("");
                                        }}
                                        style={{
                                            default: {
                                                fill: "#D6D6DA",
                                                outline: "none"
                                            },
                                            hover: {
                                                fill: "#F53",
                                                outline: "none"
                                            },
                                            pressed: {
                                                fill: "#E42",
                                                outline: "none"
                                            }
                                        }}
                                    />
                                ))
                            }
                        </Geographies>
                    </ZoomableGroup>
                </ComposableMap>
                <ReactTooltip html={true}>{props.toolTip}</ReactTooltip>
            </>
        );
    }else{
        return null;
    }

};

export default memo(CountyChart);
