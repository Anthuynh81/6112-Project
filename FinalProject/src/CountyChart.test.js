import Axios from 'axios';
import renderer from 'react-test-renderer';
import CountyChart from "./CountyChart";
import {useState} from "react";

const testDate = "03/14/2020"

async function getValue() {
    const response = await Axios.get("http://localhost:5000/County", { params: {date: testDate}})
    return response
}

test("Get request", async () => {
    const response = await getValue()
    //Success code for get request
    expect(response.status).toEqual(200)
})

test("Date filtering", async () => {
    const response = await getValue()
    //Make sure the data of the data being pulled is correct
    expect(response.data[0].Updated).toEqual(testDate)
})

test("Getting correct data", async () => {
    const response = await getValue()
    // Should be getting state data
    expect(response.data[0].AdminRegion1).toBe("Georgia")
    // Should be getting county data
    expect(response.data[0].AdminRegion2).toBe("Dougherty")
})

test('returns null when map is not county', () => {
    const component = renderer.create(
        <CountyChart setTooltipContent={null} map={"USA"} date={testDate} toolTip={null}/>
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})

test('returns the chart when map is County', () => {
    const component = renderer.create(
        <CountyChart setTooltipContent={null} map={"County"} date={testDate} toolTip={null}/>
    );
    let tree = component.toJSON()
    expect(tree.length).toBeGreaterThan(1);
})
