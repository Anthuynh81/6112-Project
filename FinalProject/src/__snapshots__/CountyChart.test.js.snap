// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns null when map is not county 1`] = `null`;

exports[`returns the chart when map is County 1`] = `
<svg
  className="rsm-svg "
  data-tip=""
  viewBox="0 0 800 600"
>
  <g>
    <rect
      fill="transparent"
      height={600}
      width={800}
    />
    <g
      className="rsm-zoomable-group undefined"
      transform="translate(0 0) scale(1)"
    >
      &gt;
      <g
        className="rsm-geographies "
      />
    </g>
  </g>
</svg>
`;
